myia/compile/backends/python/python.py

```diff
@@ -297,17 +364,68 @@ def python_universe_getitem(c, universe, handle):
     operations.getitem: operator.getitem,
     operations.bool: operator.truth,
 }
+FUNCTION_MAP = {
+    range: op_range,
+    operations.grad: op_grad,
+    operations.value_and_grad: op_value_and_grad,
+    operations.myia_next: generate_caller("__myia_next__"),
+    operations.myia_iter: generate_caller("__myia_iter__"),
+    operations.myia_hasnext: generate_caller("__myia_hasnext__"),
+    operations.user_switch: op_user_switch,
+}
 
 
-def convert_operation(c, op, *inputs):
+def convert_operation(c, node, op, *inputs):
     """Convert an operation apply node to a Python code."""
     if op.value is operations.resolve:
         namespace = inputs[0].value
         name = inputs[1].value
         resolved = namespace[name]
+        code = None
+
         if resolved in OPERATION_MAP:
-            return f"operator.{OPERATION_MAP[resolved].__name__}"
-    raise NotImplementedError(f"Unsupported operation {op}")
+            # Resolve as Python's operator module function.
+            code = f"operator.{OPERATION_MAP[resolved].__name__}"
+        elif resolved in FUNCTION_MAP:
+            # Resolve as a specific function.
+            # Function will be available as global symbol in compiled function.
+            fn = FUNCTION_MAP[resolved]
+            symbol_name = fn.__name__
+            c.register_global(symbol_name, fn)
+            code = symbol_name
+        elif isinstance(resolved, FunctionType):
+            # Resolved is a function.
+            # Function will be available as global symbol in compiled function.
+            symbol_name = resolved.__name__
+            c.register_global(symbol_name, resolved)
```

Comment:
Two different functions may have the same `__name__`, in which case they should be registered under different global names.